packages/core/src/__tests__/rum/instrumentation/DdRumUserInteractionTracking.test.tsx

```diff
@@ -185,4 +185,130 @@ it('M intercept and send a RUM event W onPress { TouchableWithoutFeedback compon
     const currentMockedEventsInterceptor = DdEventsInterceptor.mock.instances[DdEventsInterceptor.mock.instances.length - 1];
     const mockedInterceptOnPressFunction = currentMockedEventsInterceptor.interceptOnPress;
     expect(mockedInterceptOnPressFunction).toBeCalled()
-})
\ No newline at end of file
+})
+
+it('M keeps memoization working for elements W an onPress prop is passed', async () => {
+    // GIVEN
+    DdRumUserInteractionTracking.startTracking()
+    let rendersCount = 0;
+    const DummyComponent = (props) => {
+        rendersCount++;
+        return (
+            <TouchableWithoutFeedback
+                onPress={props.onPress}>
+                <View style={styles.button}>
+                    <Text style={styles.button}>Click me</Text>
+                </View>
+            </TouchableWithoutFeedback>
+        )
+    }
+    const stableOnPress = () => {};
+    const MemoizedComponent = React.memo(DummyComponent);
+    const { rerender } = render(<MemoizedComponent onPress={stableOnPress} />);
+    expect(rendersCount).toBe(1);
+
+    // WHEN
+    rerender(<MemoizedComponent onPress={stableOnPress} />);
+    // THEN
+    expect(rendersCount).toBe(1);
+
+    // WHEN
+    rerender(<MemoizedComponent onPress={() => {}} />);
+    // THEN
+    expect(rendersCount).toBe(2);
```

Comment:
I prefer adding more when/then cycles than writing a new tests when doing integration tests as they can be a bit slow to start and it makes things a bit hard to understand sometimes.
This is usually considered best practice in React: https://kentcdodds.com/blog/write-tests